File: python/euler-004.py
#A palindromic number reads the same both ways. The largest palindrome made from the product of two 2-digit numbers is 9009 = 91 × 99.
#Find the largest palindrome made from the product of two 3-digit numbers.

def palindrome(str):
    ret = True
    for i in range(len(str)):
        # 0 -1  1 -2 2 -3
        ret&=str[i]==str[i*-1 -1]
        if(ret==False):
            break
    return ret

def ran(limit):
    for i in range(limit)[::-1]:
        for x in range(limit)[::-1]:
            if(palindrome(str(x*i))):
                #print(str(x)+"*"+str(i)+"="+str(x*i))
                yield x*i

print(max(ran(1000)))

File: ruby/euler-004.rb
#A palindromic number reads the same both ways. The largest palindrome made from the product of two 2-digit numbers is 9009 = 91 × 99.
#Find the largest palindrome made from the product of two 3-digit numbers.
#906609
@max=0
def generate
  (100...999).each do |a|
    (a...999).each do |b|
      palindrome {a*b }
    end
  end
end

def palindrome
  if yield.to_s == yield.to_s.reverse
    @max = yield > @max ? yield : @max
  end
end

generate
puts @max
